django/db/backends/mysql/schema.py

```diff
@@ -34,10 +34,13 @@ def quote_value(self, value):
 
     def skip_default(self, field):
         """
-        MySQL doesn't accept default values for longtext and longblob
-        and implicitly treats these columns as nullable.
+        MySQL doesn't accept default values for TEXT and BLOB types, and
+        implicitly treats these columns as nullable.
         """
-        return field.db_type(self.connection) in {'longtext', 'longblob'}
+        return field.db_type(self.connection).lower() in {
+            'tinytext', 'tinyblob', 'text', 'blob', 'mediumtext', 'mediumblob',
+            'longtext', 'longblob',
+        }
```

Comment:
I preferred to be defensive and enumerate all the potential expected values, rather than choke on some unexpected data type that MySQL or one of its forks might have, like a `defaultacceptingtext`.